app/controllers/microservices_engine/v1/data_controller.rb

```diff
@@ -0,0 +1,51 @@
+require_dependency "microservices_engine/application_controller"
+
+module MicroservicesEngine
+  module V1
+    class DataController < ApplicationController
+
+      def index
+        # REST Endpoint
+        # Used to retrieve data from the local application
+        # Usage: http://(url)/data?models={MODELS}
+        #        MODELS: A set of model names to request data of separated
+        #                by commas.
+        # Example: http://larrysenior.com/data?models=ChewingGum,LarrysKid
+
+        objs = params.require(:models) #.permit(:attributes)
+
+        models = objs.split(',')
+        # if objs['attributes']
+        #   attrs = objs['attributes'].split(',')
+        # end
+
+        valid_model_names = MicroservicesEngine.config.endpoint_models.map{|mdl| mdl.name}
+
+        if models.all? { |param| param.in? valid_model_names }
```

Comment:
Rails comes built with ways to quickly serialize objects as JSON.

See https://github.com/umts/aggressive-epsilon/blob/master/app/controllers/v1/item_types_controller.rb#L31-L32 for an example.

You can easily do things like exclude/include certain attributes, represent nested relationships, etc.